src/App.tsx

```diff
@@ -1,26 +1,300 @@
+      }
+    }
+  };
+
+  const fetchData = async () => {
+    try {
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+    } catch {
+      Error(ErrorType.Load);
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, []);
+
+  useEffect(() => {
+    setToggleActive(todos.every(currentTodo => currentTodo.completed));
+  }, [todos]);
+
+  useEffect(() => {
+    if (shouldFocus) {
+      inputRef.current?.focus();
+    } else {
+      inputRef.current?.blur();
+    }
+  }, [dataError, todos]);
+
+  useEffect(() => {
+    setToggleActive(todos.every(currentTodo => currentTodo.completed));
+    setAllChecked(todos.every(currentTodo => currentTodo.completed));
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {todos.length !== 0 && (
+            <button
+              type="button"
+              data-cy="ToggleAllButton"
+              className={classNames(
+                'todoapp__toggle-all',
+                { active: toggleActive },
+              )}
+              onClick={() => {
+                setToggleActive(!toggleActive);
+                setUseToggle(true);
+              }}
+            />
+          )}
+
+          <form
+            onSubmit={(event) => {
+              event.preventDefault();
+              const value = inputValue.trim();
+
+              return value.length > 0
+                ? addTodo(value)
+                : Error(ErrorType.Empty);
+            }}
+          >
+            <input
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={inputValue}
+              ref={inputRef}
+              onChange={(event) => {
+                setInputValue(event.target.value);
+              }}
+              disabled={!!tempTodo}
+            />
+          </form>
+        </header>
+
+        {todos && (
+          <TodoList
+            todos={todos}
+            tempTodo={tempTodo}
+            filterType={filterType}
+            onDelete={(id: number) => deleteData(id)}
+            activeLoader={activeLoader}
+            onChange={(
+              id: number,
+              title: string,
+              completed: boolean,
+            ) => changeData(id, title, completed)}
```

Comment:
looks like we can simplify it
```suggestion
            onDelete={deleteData}
            activeLoader={activeLoader}
            onChange={changeData}
```